### PR TITLE
List tweaks

### PR DESCRIPTION
**Pull request by michalchudziak**
Sorry for the messy commit names :p

### CODE REVIEW

File: src/section-list.js [hunk 1]
```diff
@@ -79,7 +84,7 @@ export default class extends Component<Props> {
     );
 
     if (
-      y <= this._startThreshold && // nearing the top
+      y <= this.props.onStartReachedThreshold && // nearing the top
```
Comment (JakeDawkins) [inline src/section-list.js]: So wait, is this supposed to be a pt value or a ratio of the view height?

Comment (JakeDawkins) [reply src/section-list.js]: The source for SectionList says onEndReachedThreshold should be a ratio https://github.com/facebook/react-native/blob/master/Libraries/Lists/SectionList.js#L148

Comment (michalchudziak) [reply src/section-list.js]: Hmm I think `y` offset in `onScroll` call returns the value in points, we're comparing it with threshold value and event gets fired X pt's before start. 

Please correct me if I'm wrong, but within your implementation, you're doing:

```javascript
const { height } = e.nativeEvent.layout; // Height of an list

const threshold = onStartReachedThreshold ? onStartReachedThreshold : 0;
this._startThreshold = height * threshold;
```

Assuming that `onStartReachedThreshold={100}` and height of a list is `100` pt your `__startThreshold` equals 10000, and it's compared to `y` value that is always lower.

Am I missing something?

Comment (JakeDawkins) [reply src/section-list.js]: Yeah, if it was passed a large number like that, for sure. When running an example I made, I was passing it values like `0.5` for half the viewheight from the top of the list. 

The `y` offset in `onScroll` does return the value in points, but that's what the `__startThreshold` was saving.

Comment (michalchudziak) [reply src/section-list.js]: I see but I think `onEndReachedThreshold` works the way I've changed it:

https://facebook.github.io/react-native/docs/listview.html#onendreachedthreshold

> Threshold in pixels (virtual, not physical) for calling onEndReached.

Comment (JakeDawkins) [reply src/section-list.js]: I think that was for the old `ListView`, which we're not using. 

`SectionList` passes that prop straight to `VirtualizedList`, which defines it as a `unit of visible length`. 

https://github.com/facebook/react-native/blob/master/Libraries/Lists/VirtualizedList.js#L139

Also, we have a value for `20` set for the `onEndReachedThreshold` in Darwin right now, and it fires _much_ earlier than 20pt from the bottom.

https://github.com/majorleaguesoccer/darwin/blob/master/screens/schedule-hub/schedule.js#L109

**Comment by michalchudziak** [comment]
I'm fine with keeping threshold how it was before. I think you're right with the ratio - I didn't know it has changed in SectionList implementation. IMO it's much more confusing, but definitely we should do the same as  native `onEndReachedThreshold`.

**Comment by JakeDawkins** [comment]
I agree! I'd rather have px, but we'll just have to live with this. Haha, thanks! I'll get it updated right now

**Pull request merged**